src/test/java/de/tum/cit/aet/artemis/iris/IrisChatSessionResourceTest.java

```diff
@@ -0,0 +1,181 @@
+
+    @Autowired
+    private LectureUtilService lectureUtilService;
+
+    private Lecture lecture;
+
+    private Course course;
+
+    @BeforeEach
+    void initTestCase() throws Exception {
+        List<User> users = userUtilService.addUsers(TEST_PREFIX, 3, 1, 0, 1);
+        List<Course> courses = courseUtilService.createCoursesWithExercisesAndLectures(TEST_PREFIX, true, true, 1);
+        course = courses.getFirst();
+        textExercise = textExerciseUtilService.createIndividualTextExercise(course, ZonedDateTime.now(), ZonedDateTime.of(2040, 9, 9, 12, 12, 0, 0, ZoneId.of("Europe/Berlin")),
+                ZonedDateTime.of(2040, 9, 9, 12, 12, 0, 0, ZoneId.of("Europe/Berlin")));
+        StudentParticipation studentParticipation = new StudentParticipation().exercise(textExercise);
+        studentParticipation.setParticipant(userUtilService.getUserByLogin(TEST_PREFIX + "student1"));
+        studentParticipationRepository.save(studentParticipation);
+
+        soloExercise = ExerciseUtilService.getFirstExerciseWithType(course, ProgrammingExercise.class);
+        teamExercise = programmingExerciseUtilService.addProgrammingExerciseToCourse(course);
+        teamExercise.setMode(ExerciseMode.TEAM);
+        programmingExerciseRepository.save(teamExercise);
+
+        Team team = new Team();
+        team.setName("Team 1");
+        team.setShortName("team1");
+        team.setExercise(teamExercise);
+        team.setStudents(Set.of(users.get(1), users.get(2)));
+        team.setOwner(users.get(1));
+        final var savedTeam = teamRepository.save(team);
+
+        pipelineDone = new AtomicBoolean(false);
+
+        lecture = lectureUtilService.createLecture(course, ZonedDateTime.now());
+        activateIrisGlobally();
+        activateIrisFor(course);
+        activateIrisFor(textExercise);
+        activateIrisFor(teamExercise);
+        activateIrisFor(soloExercise);
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
+    void getAllSessionsForCourseWithoutSessions() throws Exception {
+        List<IrisChatSession> irisChatSessions = request.getList("/api/iris/chat-history/" + course.getId() + "/sessions", HttpStatus.OK, IrisChatSession.class);
+        assertThat(irisChatSessions).hasSize(0);
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
+    void getSessionForSessionId() throws Exception {
+        var courseSession = createCourseChatSessionForUser("student1");
+        IrisSession irisChatSessions = request.get("/api/iris/chat-history/" + course.getId() + "/session/" + courseSession.getId(), HttpStatus.OK, IrisSession.class);
+        assertThat(irisChatSessions.getId()).isEqualTo(courseSession.getId());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
+    void getAllSessionsForCourseWithSessions() throws Exception {
+        var courseSession = createCourseChatSessionForUser("student1");
```

Comment:
you are using a lot of var assignments in this test. Is it possible to use the actuall Types instead?